docs/charts_tips_and_tricks.md

```diff
@@ -96,6 +96,36 @@ For example:
 The above will render the template when .Values.foo is defined, but will fail
 to render and exit when .Values.foo is undefined.
 
+## Creating Image Pull Secrets
+Image pull secrets are essentially a combination of _registry_, _username_, and _password_.  You may need them in an application you are deploying, but to create them requires running base64 a couple of times.  A helper can help you do this.  Here is an example:
+
+Presuming that the values are provided in the values.yaml like so
+```
+imageCredentials:
+  registry: quay.io
+  username: someone
+  password: sillyness
+```  
+
+You can add a helper like so:
+
+```
+{{- define "imagePullSecret" }}
+{{- printf "{\"auths\": {\"%s\": {\"auth\": \"%s\"}}}" .Values.imageCredentials.registry (printf "%s:%s" .Values.imageCredentials.username .Values.imageCredentials.password | b64enc) | b64enc }}
```

Comment:
- [ ] This format is different from the one that I've used locally, which uses a top-level object key for the server URL, with a corresponding value object with keys "username," "email," "password," and "auth." That sort of matches your format here, but without the top-level "auths" field. 
  Where did you find this format used here?

Reply:
Its just stuff I've created on my own.  username/password is somewhat mutually exclusive from auth I believe.  I guess we could add a more complicated example that switches on existence of auth.

Part of this tip/trick was to show someone how to do nested printf statements, which wasn't very clear to me when I first started using helm.  I actually contemplated having the example set the scope to `.Values.imageCredentials` instead of `.` (and thus using `.username` and so) so that it might serve as another example of how to set template scope.

Arguably an application could pull from different registries (different containers in a pod) and have multiple deployments (with different credentials) - so it might be a bit presumptive to do it the way you suggested (as there may be two credentials in play from the same registry, if I'm understanding you correctly)

For me it just made sense to offer it this way - if it was meant for a app-wide image pull secret, or settings for a specific image (since imagePullSecrets can be an array for a pod)

Let me know what you think - I can change it - again I'm more interested in helping people understand how to do tasks like this than telling someone how to arrange their `values.yaml` file.